docs/how-to-guides/meson-args.rst

```diff
@@ -17,5 +16,125 @@ Passing arguments to Meson_
    - Mention the order in which Meson arguments are added, and how it affect
      Meson
 
+Advanced meson options can be accessed by specifying extra arguments to the
+individual meson commands:
+
+   - meson dist: https://mesonbuild.com/Commands.html#dist
+   - meson setup: https://mesonbuild.com/Commands.html#setup
+   - meson compile: https://mesonbuild.com/Commands.html#compile
+   - meson install: https://mesonbuild.com/Commands.html#install
+
+These arguments can be added to the project by adding a section
+to the project's ``pyproject.toml``, or as arguments when building
+the package using the :ref:`build config settings<how-to-guides-config-settings>`
+
+.. tab-set::
+
+    .. tab-item:: pyproject.toml
+        :sync: key_pyproject_toml
+
+		.. code-block::
+
+			[tool.meson-python.args]
+			dist    = ['dist-argument_1', 'dist-argument_2', '...']
+			setup   = ['setup-argument_1', 'setup-argument_2', '...']
+			compile = ['compile-argument_1', 'compile-argument_2', '...']
+			install = ['install-argument_1', 'install-argument_2', '...']
+
+    .. tab-item:: pypa/buid
+        :sync: key_pypa_build
+
+		.. code-block:: console
+
+			$ python -m build -Cdist-args="args"    \
+					  -Csetup-args="args"   \
+					  -Ccompile-args="args" \
+					  -Cinstall-args="args" .
+
+    .. tab-item:: pip
+        :sync: key_pip
+
+		.. code-block:: console
+
+			$ python -m pip --config-settings=dist-args="args"    \
+					--config-settings=setup-args="args"   \
+					--config-settings=compile-args="args" \
+					--config-settings=install-args="args" .
+
+
+Example 1: set the default libraries to *static*
+================================================
+
```

Comment:
Can we have an "Example" section and then "1) Set the ..." and "2) Use Meson ..." subsections? IMO that'd look better.